add deleteHook API call

zzo opened the pull request:
add the delete hook call - thanks for the great package!

pksunkara commented:
Please edit the coffee script files in the src folder and then generate the javascript in lib folder.

zzo commented:
Ha!  right sorry didn't notice that :)
I've updated repo.coffee
Unfortunately(?) there's a newer version of coffee script so when I generated the JS files in lib they were updated with the latest version in the comment - dunno if you want to keep that or not (it's going to happen eventually anyway) - but up to you.
Also updated the REAMDE

pksunkara commented on src/octonode/repo.coffee:
```diff
@@ -384,6 +384,13 @@ class Repo
       if s isnt 201 then cb(new Error("Repo createHook error")) else cb null, b, h
 
+  # Delete a hook
+  # '/repos/pksunkara/hub/hooks/hookId' DELETE
```
Put `37` instead of `hookId` in this line.

pksunkara commented on src/octonode/repo.coffee:
```diff
@@ -384,6 +384,13 @@ class Repo
 
+  # Delete a hook
+  # '/repos/pksunkara/hub/hooks/hookId' DELETE
+  deleteHook: (hookId, cb) ->
```
Use the variable name `id` instead of `hookId` in this function.

pksunkara commented on README.md:
````diff
@@ -618,6 +618,12 @@ ghrepo.hook({
 }, callback); // hook
 ```
 
+#### Delete a hook (DELETE /repos/pksunkara/hub/hooks/hookId)
````
Use `37` instead of `hookId` in this line.

pksunkara commented:
Looks good to me. Just a few minor styling fixes needed. No problem with the coffee-script version upgrade.

zzo commented:
done thanks!

pksunkara commented:
Looks good. Could you squash all the commits into a single commit? I don't want to unnecessarily complicate the git logs. Thanks

zzo commented:
sorry my git foo is weak I don't know why there's that empty commit there

pksunkara commented:
You see that there are 8 commits in here, right? Can you rebase them all to a single commit? Use git interactive rebase.

zzo commented:
thanks I did that to get commit https://github.com/zzo/octonode/commit/c67a4c47c78947d0d801d1db805ae47842813af8
maybe i screwed something up

pksunkara commented:
Thanks for the PR. I really appreciate these contributions. :smile:
